proxy/http/remap/PluginFactory.cc

```diff
@@ -236,25 +244,44 @@ PluginFactory::findByEffectivePath(const fs::path &path)
 }
 
 /**
- * @brief Tell all plugins (that so wish) that remap.config is being reloaded
+ * @brief Tell all plugins instantiated by this factory that the configuration
+ * they are using is no longer the active one.
  *
  * This method would be useful only in case configs are reloaded independently from
  * factory/plugins instantiation and initialization.
  */
 void
-PluginFactory::indicateReload()
+PluginFactory::deactivate()
 {
-  Debug(_tag, "indicated config reload to factory '%s'", getUuid());
+  Debug(_tag, "deactivate configuration used by factory '%s'", getUuid());
 
   _instList.apply([](RemapPluginInst &pluginInst) -> void { pluginInst.done(); });
+}
 
-  _list.apply([](PluginDso &plugin) -> void {
-    if (1 == plugin.instanceCount()) {
-      plugin.done();
-    } else {
-      plugin.indicateReload();
-    }
-  });
+/**
+ * @brief Tell all plugins (that so wish) that remap.config is going to be reloaded
+ */
+void
+PluginFactory::indicatePreReload()
+{
+  Debug(_tag, "indicated config is going to be reloaded by factory '%s' to %zu plugin%s", getUuid(), _list.count(),
+        _list.count() != 1 ? "s" : "");
+
+  _list.apply([](PluginDso &plugin) -> void { plugin.indicatePreReload(); });
+}
+
+/**
+ * @brief Tell all plugins (that so wish) that remap.config is done reloading
+ */
+void
+PluginFactory::indicatePostReload(TSReturnCode reloadStatus)
+{
+  Debug(_tag, "indicated config is done reloading by factory '%s' to %zu plugin%s", getUuid(), _list.count(),
+        _list.count() != 1 ? "s" : "");
+
+  for (auto it = _list.begin(); _list.end() != it; it++) {
```

Comment:
Why not use `apply` here, or even just a range `for` loop? E.g.
```
for ( auto & inst : _list ) { inst->indicatePostReload(reloadStatus); }
```

Reply:
Sure, can do the "eye-candy" as well. I have stared at green monochrome monitors and this now-outdated idiom for so long so "[I don't even see the code. All I see is blonde, brunette, redhead](https://www.youtube.com/watch?v=3vAnuBtyEYE)"... I also liked `apply` but enhancing it did not feel essential for this change, simple looping over the objects looks concise enough to me.